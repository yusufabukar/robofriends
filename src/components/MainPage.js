import React, { Component} from 'react';
import Header from './Header';
import CountButton from './CountButton';
import SearchBox from './SearchBox';
import ErrorBoundary from './ErrorBoundary';
import CardList from './CardList';
import Scroll from './Scroll';
// import { robots } from './robots';
import './MainPage.css';

class MainPage extends Component {
    componentDidMount() {
        this.props.onRequestRobots();
    };

    filterRobots = () => {
        return this.props.robots.filter(robot => {
            return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
        });
    };

    render() {
        const { isPending, robots, onSearchChange } = this.props;

        return isPending ?
            <h1 className='tc'>Loading</h1> :
        (
            <div className='tc'>
                <Header />
                <CountButton />
                <SearchBox searchChange={onSearchChange}/>
                <Scroll>
                    <ErrorBoundary>
                        <CardList robots={this.filterRobots()}/>
                    </ErrorBoundary>
                </Scroll>
            </div>
        );
    };
}

export default MainPage;