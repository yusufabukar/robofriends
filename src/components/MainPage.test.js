import React from 'react';
import { shallow } from 'enzyme';
import MainPage from './MainPage';

let wrapper;
beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        searchField: '',
        robots: [],
        isPending: false
    };
    wrapper = shallow(<MainPage {...mockProps} />);
});

it('Renders MainPage', () => {
    expect(wrapper).toMatchSnapshot();
});

it('Filters robots correctly', () => {
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        searchField: 'JA',
        robots: [{
            id: 4,
            name: 'Jason',
            email: 'juju@gmail.com'
        }],
        isPending: false
    };

    const wrapper2 = shallow(<MainPage {...mockProps2} />);
    expect(wrapper2.instance().filterRobots()).toEqual([{
        id: 4,
        name: 'Jason',
        email: 'juju@gmail.com'
    }]);
});

it('Filters robots correctly 2', () => {
    const mockProps3 = {
        onRequestRobots: jest.fn(),
        searchField: 'YS',
        robots: [{
            id: 7,
            name: 'Jason',
            email: 'juju@gmail.com'
        }],
        isPending: false
    };

    const filteredRobots = [];
    const wrapper3 = shallow(<MainPage {...mockProps3} />);
    expect(wrapper3.instance().filterRobots()).toEqual(filteredRobots);
});