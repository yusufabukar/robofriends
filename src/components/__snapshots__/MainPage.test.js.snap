// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders MainPage 1`] = `
<div
  className="tc"
>
  <Header />
  <CountButton />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
